django/core/checks/urls.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import unicode_literals
+
+from . import Tags, Warning, register
+
+
+@register(Tags.urls)
+def check_url_config(app_configs, **kwargs):
+    from django.core.urlresolvers import get_resolver
+    resolver = get_resolver()
+    return check_resolver(resolver)
+
+
+def check_resolver(resolver):
+    """
+    Recursively check the resolver
+    """
+    from django.core.urlresolvers import RegexURLPattern, RegexURLResolver
+    warnings = []
+    for pattern in resolver.url_patterns:
+        if isinstance(pattern, RegexURLResolver):
+            warnings.extend(check_include_trailing_dollar(pattern))
+            # Check resolver recursively
+            warnings.extend(check_resolver(pattern))
+        elif isinstance(pattern, RegexURLPattern):
+            warnings.extend(check_pattern_name(pattern))
+
+        warnings.extend(check_pattern_startswith_slash(pattern))
+
+    return warnings
+
+
+def describe_pattern(pattern):
+    """
+    Formats the URL pattern for display in warning messages
+    """
+    description = pattern.regex.pattern
+    if getattr(pattern, 'name', False):
+        description += " [name='{}']".format(pattern.name)
+    return description
+
+
+def check_include_trailing_dollar(pattern):
+    """
+    Checks that include is not used with a regex ending with a dollar.
+    """
+    regex_pattern = pattern.regex.pattern
+    if regex_pattern.endswith('$') and not regex_pattern.endswith('\$'):
+        warning = Warning(
+            "Your URL pattern {} uses include with a regex ending with a '$'. "
```

Comment:
I think the formatted pattern should be wrapped in quotes. "Your URL pattern '{}' uses..".